README.md

```diff
@@ -6,35 +6,86 @@ A Swift SignalR Client for the Asp.Net Core version of SignalR
 
 ### Cocoapods
 
-- add the following lines to your `Podfile`:
-  ```ruby
-  use_frameworks!
-  pod 'SwiftSignalRClient'
-  ```
-- run:
-  ```
-  pod install
-  ```
+Add the following lines to your `Podfile`:
 
-Add `#import SwiftSignalRClient` to swift files you would like to use the client in.
-
-## Samples
+```ruby
+use_frameworks!
+pod 'SwiftSignalRClient', '~> 0.5'
+```
 
-The repo contains samples for:
+Then run:
+```sh
+pod install
+```
 
-  - [Hubs](https://github.com/moozzyk/SignalR-Client-Swift/tree/master/HubSample)
-  - [HttpConnection](https://github.com/moozzyk/SignalR-Client-Swift/tree/master/ConnectionSample)
+### Swift Packacge Manager
 
-The samples require a running server. To start the server go to the `TestServer` and 
+Add the following to your `Package` dependencies:
 
-- install the JS client running (needed only once):
+```swift
+.package(url: "https://github.com/moozzyk/SignalR-Client-Swift", .upToNextMinor(from: "0.5.0")),
 ```
-npm install
+
+Then include `"SwiftSignalRClient"` in your target dependencies. For example:
+
+```swift
+.target(name: "MySwiftPackage", dependencies: ["SwiftSignalRClient"]),
 ```
-- start the server with: 
 
-```C#
-dotnet run
+## Usage
+
+Add `import SwiftSignalRClient` to swift files you would like to use the client in.
+
+A typical implementation looks like the following:
+
+```swift
+import Foundation
+import SwiftSignalRClient
+
+public class SignalRService {
+    private var connection: HubConnection
+    
+    public init(url: URL) {
+        connection = HubConnectionBuilder(url: url).withLogging(minLogLevel: .error).build()
+        connection.on(method: "MessageReceived", callback: { (args, typeConverter) in
+            do {
+                let user = try typeConverter.convertFromWireType(obj: args[0], targetType: String.self)
+                let message = try typeConverter.convertFromWireType(obj: args[1], targetType: String.self)
+                self.handleMessage(message, from: user)
+            } catch {
+                print(error)
+            }
+        })
+        
+        connection.start()
+    }
+    
+    private func handleMessage(_ message: String, from user: String) {
+        // Do something with the message.
+    }
+}
 ```
 
-(Requires [.NET Core SDK 2.1.300](https://www.microsoft.com/net/download/dotnet-core/sdk-2.1.300) or later)
+## Examples
+
+There are several sample projects in the `Examples` folder. They include:
+
+  - [SignalRClient.xcworkspace](Examples/)
+    
+    An Xcode workspace that has compiled libraries for macOS (OSX) and iOS, along with the Application targets 'ConnectionSample', 'HubSample', and 'HubSamplePhone'.
+    
+  - [TestServer](Examples/TestServer)
```

Comment:
TestServer is a bit weird. Samples do use it but it's primarily there for tests (I just did not feel a need to create two separate servers). I am not sure what are SPM requirements/conventions but was wondering if TestServer should not be moved to Tests folder.

Reply:
I'm not aware of any requirements around testing with SPM, but I would argue the case that the 'Tests' directory only be for the test targets and the code directly testing the library functions. (Yes it's true that you need the server running in order for the tests to run/succeed, but that also requires you to have a .Net environment setup in order to run the solution.) If you feel particularly strong one way or the other, I'd be happy to make the changes.

Reply:
This is fine - I don't really see one way to be significantly better (or better at all) than the other. Let's leave it as is for now and it always can be changed in the future if needed.